src/main/java/org/apache/maven/plugins/jar/AbstractJarMojo.java

```diff
@@ -196,16 +196,17 @@ protected File getJarFile( File basedir, String resultFinalName, String classifi
             throw new IllegalArgumentException( "finalName is not allowed to be null" );
         }
 
-        StringBuilder fileName = new StringBuilder( resultFinalName );
-
+        final String fileName;
```

Comment:
not sure what is the need if final here?

Reply:
No real need, it's mostly just syntactic sugar in this case. It's somewhat a statement of intent indicating that the value will be calculated only once.
This was the kind of feedback I was seeking though - if you're happy with the changeset in principle, I'll take the `final`s out, cut a Jira ticket and make the PR more presentable etc.

Reply:
all good. please remove those `final` as I don't think they make sense for a local variable. except adding more useless code.
btw I'm more a sugar free person if it's not really needed 🤣 
if you want to create a jira why not but I'm not sure we need a Jira for such code polishing.
it's up2u

Reply:
The less I need to touch Jira the better. OK, I've removed the `final`s and updated the PR description

Reply:
@rhowe great. Thanks! Still on final to remove  ;)

Reply:
Oops, OK the final `final` is gone